alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportDocLevelMonitorFanOutAction.kt

```diff
@@ -0,0 +1,1186 @@
+                    null
+                )
+                val getDestinationsRequest = GetDestinationsRequest(
+                    action.destinationId,
+                    0L,
+                    null,
+                    table,
+                    "ALL"
+                )
+
+                val getDestinationsResponse: GetDestinationsResponse = client.suspendUntil {
+                    client.execute(GetDestinationsAction.INSTANCE, getDestinationsRequest, it)
+                }
+                getDestinationsResponse.destinations.firstOrNull()
+            } catch (e: IllegalStateException) {
+                // Catching the exception thrown when the Destination was not found so the NotificationActionConfigs object can be returned
+                null
+            } catch (e: OpenSearchSecurityException) {
+                if (notificationPermissionException != null)
+                    throw notificationPermissionException
+                else
+                    throw e
+            }
+
+            if (destination == null && notificationPermissionException != null)
+                throw notificationPermissionException
+        }
+
+        return NotificationActionConfigs(destination, channel)
+    }
+
+    private fun getDestinationContextFactory(): DestinationContextFactory {
+        val destinationSettings = DestinationSettings.loadDestinationSettings(settings)
+        return DestinationContextFactory(client, xContentRegistry, destinationSettings)
+    }
+
+    private fun compileTemplate(template: Script, ctx: TriggerExecutionContext): String {
+        return scriptService.compile(template, TemplateScript.CONTEXT)
+            .newInstance(template.params + mapOf("ctx" to ctx.asTemplateArg()))
+            .execute()
+    }
+
+    private suspend fun onSuccessfulMonitorRun(monitor: Monitor) {
+        alertService.clearMonitorErrorAlert(monitor)
+        if (monitor.dataSources.alertsHistoryIndex != null) {
+            alertService.moveClearedErrorAlertsToHistory(
+                monitor.id,
+                monitor.dataSources.alertsIndex,
+                monitor.dataSources.alertsHistoryIndex!!
+            )
+        }
+    }
+
+    private fun constructErrorMessageFromTriggerResults(
+        triggerResults: MutableMap<String, DocumentLevelTriggerRunResult>? = null
+    ): String {
+        var errorMessage = ""
+        if (triggerResults != null) {
+            val triggersErrorBuilder = StringBuilder()
+            triggerResults.forEach {
+                if (it.value.error != null) {
+                    triggersErrorBuilder.append("[${it.key}]: [${it.value.error!!.userErrorMessage()}]").append(" | ")
+                }
+            }
+            if (triggersErrorBuilder.isNotEmpty()) {
+                errorMessage = "Trigger errors: $triggersErrorBuilder"
+            }
+        }
+        return errorMessage
+    }
+
+    internal fun isActionActionable(action: Action, alert: Alert?): Boolean {
```

Comment:
is this method duplicated? can it be re-used across monitors instead of duplicating the code

Reply:
removed duplication.